src/Sylius/Bundle/ResourceBundle/EventListener/ODMMappedSuperClassSubscriber.php

```diff
@@ -76,17 +76,25 @@ private function convertToDocumentIfNeeded(ClassMetadataInfo $metadata)
     private function setAssociationMappings(ClassMetadataInfo $metadata, $configuration)
     {
         foreach (class_parents($metadata->getName()) as $parent) {
-            $parentMetadata = new ClassMetadata($parent);
-
-            if (false === $this->isSyliusClass($parentMetadata)) {
+            if (false === in_array($parent, $configuration->getMetadataDriverImpl()->getAllClassNames())) {
                 continue;
             }
 
-            if (false === in_array($parent, $configuration->getMetadataDriverImpl()->getAllClassNames())) {
+            $parentMetadata = new ClassMetadata(
+                $parent,
+                $configuration->getNamingStrategy()
```

Comment:
This was my only question. Do the tests check ODM as well? This line wasn't here before in ODM only in ORM